Mathlib/Algebra/MonoidAlgebra/Division.lean

```diff
@@ -200,6 +225,44 @@ theorem of'_dvd_iff_modOf_eq_zero {x : k[G]} {g : G} :
     exact dvd_mul_right _ _
 #align add_monoid_algebra.of'_dvd_iff_mod_of_eq_zero AddMonoidAlgebra.of'_dvd_iff_modOf_eq_zero
 
+lemma mul_divOf (x y : k[G]) (g : G) :
+    (x * y) /ᵒᶠ g =
+    of' k G g * (x /ᵒᶠ g) * (y /ᵒᶠ g) +
+    (x /ᵒᶠ g) * (y %ᵒᶠ g) +
+    (x %ᵒᶠ g) * (y /ᵒᶠ g) +
+    ((x %ᵒᶠ g) * (y %ᵒᶠ g)) /ᵒᶠ g := by
+  rw [← congr_arg₂ HMul.hMul (divOf_add_modOf x g) (divOf_add_modOf y g), add_mul, mul_add, mul_add,
+    add_divOf, add_divOf, add_divOf, mul_assoc, of'_mul_divOf, ← mul_assoc,
+    ← of'_commute (.all _) _, add_assoc, add_assoc, add_assoc]
+  congr 1
+  rw [mul_assoc, of'_mul_divOf]
+  congr 1
+  rw [(of'_commute (.all _) _).symm.left_comm, of'_mul_divOf]
+
+lemma mul_modOf (x y : k[G]) (g : G) :
+    (x * y) %ᵒᶠ g = ((x %ᵒᶠ g) * (y %ᵒᶠ g)) %ᵒᶠ g := by
+  have := congr_arg₂ HMul.hMul (divOf_add_modOf x g) (divOf_add_modOf y g)
+  simp only [mul_add, add_mul, (of'_commute (AddCommute.all g) _).symm.left_comm] at this
+  rw [← this]
+  simp_rw [add_modOf, mul_assoc, of'_mul_modOf, zero_add]
+
+/-- `hs` is a more general spelling of `¬(s ≤ a) → ¬(s ≤ b) → ¬(s ≤ a + b)`. -/
+lemma mul_modOf_of_minimal (x y : k[G]) (s : G)
+    (hs : ∀ a b, ¬(∃ d, a = s + d) → ¬(∃ d, b = s + d) → ¬(∃ d, a + b = s + d)) :
```

Comment:
@YaelDillies, is there a higher-level description of what this statement means? I'm pretty sure this is the correct generality for this file, but a more intuitive docstring would be nice. The intended application is `G := σ →₀ ℕ`, where `s := Finsupp.single` is a suitable choice of `S`; or `G := ℕ`, where `s = 1` is such a choice.

Is it "covers zero"?

Reply:
I agree with "covers zero" in the case where `G` is canonically ordered. In general this sounds more like "order-inaccessible by addition"? kind of like a compact element, except we take addition instead of supremum.